lib/sqlalchemy/sql/elements.py

```diff
@@ -4665,14 +4805,16 @@ def _from_objects(self) -> List[FromClause]:
             return []
 
     @HasMemoized.memoized_attribute
-    def _render_label_in_columns_clause(self):
+    def _render_label_in_columns_clause(self) -> bool:
         return self.table is not None
 
     @property
-    def _ddl_label(self):
+    def _ddl_label(self) -> Optional[str]:
         return self._gen_tq_label(self.name, dedupe_on_key=False)
 
-    def _compare_name_for_result(self, other):
+    def _compare_name_for_result(
+        self, other: ColumnElement[Any]
+    ) -> Union[bool, FrozenSet[ColumnElement[Any]]]:
```

Comment:
Returns `bool` normally, has to return `Union[bool, FrozenSet]` here. Writing a single `@overload` makes mypy complain about it. Could be solved by adding the `Union` to all 3 definitions, but that's a bit misleading - there is even a comment saying specifically that the function returns `bool`. Not sure how to proceed

Reply:
Well, I'm afraid in such case there are only 2 ways. Either split the method onto 2 different methods (it looks like being a part of private API, so then you'd have to update all of the internal calls to it) or update parent signatures like that:
```diff
diff --git a/lib/sqlalchemy/sql/elements.py b/lib/sqlalchemy/sql/elements.py
index 10a63ecc4..49b18aefa 100644
--- a/lib/sqlalchemy/sql/elements.py
+++ b/lib/sqlalchemy/sql/elements.py
@@ -1537,7 +1537,7 @@ class ColumnElement(
 
         return bool(self.proxy_set.intersection(othercolumn.proxy_set))
 
-    def _compare_name_for_result(self, other: ColumnElement[Any]) -> bool:
+    def _compare_name_for_result(self, other: ColumnElement[Any]) -> Union[bool, FrozenSet[ColumnElement[Any]]]:
         """Return True if the given column element compares to this one
         when targeting within a result row."""
 
@@ -4382,7 +4382,7 @@ class NamedColumn(KeyedColumnElement[_T]):
     name: str
     key: str
 
-    def _compare_name_for_result(self, other: ColumnElement[Any]) -> bool:
+    def _compare_name_for_result(self, other: ColumnElement[Any]) -> Union[bool, FrozenSet[ColumnElement[Any]]]:
         return (hasattr(other, "name") and self.name == other.name) or (
             hasattr(other, "_label") and self._label == other._label
         )

```